tests/readme.md

```diff
@@ -55,44 +54,108 @@ examples.
 To run all tests, run `npm install` and then run:
 
 ```bash
-npm run test
+npm test
 ```
 
 To run an individual test, run:
 
 ```bash
-npm run test ${path_to_linter_subdir}
+npm test ${path_to_linter_subdir}
 ```
 
-Then verify that the generated snapshot file includes the results you would expect (e.g. several
-fileIssues, no taskFailures).
+Then verify that the generated snapshot file includes the results you would expect (e.g. an Object
+with several fileIssues, no taskFailures).
 
 ### Linter Versioning
 
 Missing snapshots will be automatically created based on test input files. If an existing test fails
-because a new linter version has introduced a breaking change, rather than running
-`npm run test -- -u`, **instead run**
-`PLUGINS_TEST_NEW_SNAPSHOT=true npm run test ${path_to_failing_test}`. This is used to track
-historical test behavior and ensure compatibility with trunk across multiple linter versions. See
-"Environment Overrides" below.
+because a new linter version has introduced a breaking change, rather than running `npm test -- -u`,
```

Comment:
```suggestion
because a new linter version has introduced a breaking change, rather than running `npm test -- -u`,
```

Can we detect `-u` and fail?

If not, NBD.

Reply:
We can... but -u should also be an option, in case we need to update an old snapshot (should be rare, but can and will happen. This is also useful for debugging if you've configured something poorly; e.g. your test runs check, and it generates a taskFailure that you did not intend (right now we don't error out on the presence of taskFailures, in the future we may want to add more test helper functions than just `linterCheckTest` that assert these kinds of failures, such as on empty files).

Overall, we may want to revisit some of the specifics related to this particular workflow in the future, particularly as we see any pitfalls that users are hitting when contributing.